lib/Sema/SemaDecl.cpp

```diff
@@ -13083,9 +13083,13 @@ ExprResult Sema::VerifyBitField(SourceLocation FieldLoc,
     // Handle incomplete types with specific error.
     if (RequireCompleteType(FieldLoc, FieldTy, diag::err_field_incomplete))
       return ExprError();
-    if (FieldName)
-      return Diag(FieldLoc, diag::err_not_integral_type_bitfield)
-        << FieldName << FieldTy << BitWidth->getSourceRange();
+    if (FieldName) {
+      if (getLangOpts().CheckedC && FieldTy->isPointerType())
+        return Diag(BitWidth->getExprLoc(), diag::err_expected_bounds_expr_for_member);
```

Comment:
clang tries to avoid printing out redundant information when source information already has that information.  In this case, the error message will point directly at the beginning of the offending bounds expression or bit-width expression.   I don't think we need add more information here.